### PR TITLE
Add additional tracker events

### PR DESCRIPTION
**Pull request by yubozhao**

### CODE REVIEW

File: bentoml/archive/loader.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import tempfile
 import logging
 
-from bentoml.utils.usage_stats import track_loading
+from bentoml.utils.usage_stats import track_loading, track_load_start
```
Comment (parano) [inline bentoml/archive/loader.py]: suggest rename: track_loading => track_load_finish, no need to change the actual tag in amptitude

File: bentoml/server/bento_api_server.py [hunk 2]
```diff
@@ -289,4 +290,6 @@ def start(self):
         """
         Start an REST server at the specific port on the instance or parameter.
         """
+        track_server('flask')
```
Comment (parano) [inline bentoml/server/bento_api_server.py]: this event will be duplicated when running gunicorn server right?

File: bentoml/utils/usage_stats.py [hunk 3]
```diff
@@ -86,6 +87,10 @@ def track_save(bento_service):
         return track("save", info)
 
 
+def track_load_start():
+    return track('load-start', {})
```
Comment (parano) [inline bentoml/utils/usage_stats.py]: let's move the logic of checking `if config['core'].getboolean("usage_tracking"):`  to track function above :)

Comment (parano) [reply bentoml/utils/usage_stats.py]: otherwise it's easy to forget

File: bentoml/utils/usage_stats.py [hunk 4]
```diff
@@ -100,6 +105,19 @@ def track_cli(command, deploy_platform=None):
         return track('cli-' + command, info)
 
 
+def track_server(server_type):
+    start_time = time.time()
+    import atexit
```
Comment (parano) [inline bentoml/utils/usage_stats.py]: import at the top of the file

File: bentoml/server/bento_api_server.py (hunk 2 above)
Comment (parano) [reply bentoml/server/bento_api_server.py]: anyway to workaround?

Comment (yubozhao) [reply bentoml/server/bento_api_server.py]: It is duplicated in each workers for gunicorn. I thought about putting it at server initialization. However I really want uptime for it, so it ends up on each worker. 

It’s is bad send the message multiple times. While I think it is kinda neat data point for how many workers

Comment (yubozhao) [reply bentoml/server/bento_api_server.py]: Open for suggestions though

File: bentoml/utils/usage_stats.py (hunk 3 above)
Comment (yubozhao) [reply bentoml/utils/usage_stats.py]: Yep. Good catch

File: bentoml/archive/loader.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 import tempfile
 import logging
 
-from bentoml.utils.usage_stats import track_loading
+from bentoml.utils.usage_stats import track_loading_finish, track_load_start
```
Comment (parano) [inline bentoml/archive/loader.py]: nit: s/track_loading_finish/track_load_finish/ just to be consistent

File: bentoml/server/gunicorn_server.py [hunk 6]
```diff
@@ -82,3 +83,7 @@ def load(self):
         bento_service = load(self.bento_archive_path)
         api_server = BentoAPIServer(bento_service, port=self.port)
         return api_server.app
+
+    def run(self):
+        track_server('gunicorn')
```
Comment (parano) [inline bentoml/server/gunicorn_server.py]: just add an attribute "num_of_workers" here in the amplitude message, you can access self. num_of_workers

File: bentoml/server/bento_sagemaker_server.py [hunk 7]
```diff
@@ -61,4 +62,5 @@ def __init__(self, bento_service, api_name, app_name=None):
         setup_routes(self.app, self.bento_service, api_name)
 
     def start(self):
+        track_server('sagemaker')
```
Comment (parano) [inline bentoml/server/bento_sagemaker_server.py]: notice the typo in comment above, "..an AWS sagemaker compatibility *REST* server."

**Pull request merged**